emailApp created for apscheduler

jiaqi1993 opened the pull request:
**Why**
Created emailApp for APScheduler to use

<!-- How does it address the problem? -->

**How**
Check if CMSStatus ID 1, cms.active==True and time_difference==True,
Send email 

<!-- Are there any side effects? -->

**Side Effects**
All condition testing works locally.

**Pending to be tested before running scheduler on heroku**
1. To create a CMSStatus id=1 in my own database. 
2. For the scheduler to work, "sendemail: python sendemail.py" needs to be put in the Procfile

jiaqi1993 committed:
emailApp created for apscheduler

jiaqi1993 committed:
Update changes to sendemail.py

chinjieh commented on emailApp/views.py:
```diff
@@ -0,0 +1,54 @@
+	 	cms.active = json_obj.get("active")
+	 	cms.last_sent = json_obj.get("last_sent")
+
+	 	time_difference = (datetime_util.sgt_now() - cms.last_sent) > timedelta(minutes = 30)
```
`time_difference` is a boolean but looks like a float / number from its name! I suggest `time_exceeds_interval` instead, or _something_ that looks like it makes sense in and `if` statement: imagine this

```
if time_exceeds_interval: 
    do stuff
```

That will look really nice and clean.

Just encouraging good Pythonic coding practices...

chinjieh commented on emailApp/views.py:
```diff
@@ -0,0 +1,54 @@
+	 	cms.last_sent = json_obj.get("last_sent")
+
+	 	time_difference = (datetime_util.sgt_now() - cms.last_sent) > timedelta(minutes = 30)
+	 	if (time_difference == True and cms.active == True):
```
I think for a cleaner, more Pythonic way, can then write:

```
if time_exceeds_interval and cms.active:
```

See how the name change mentioned above helps in readability! Why I like Python...

chinjieh commented on emailApp/untitled:
```diff
@@ -0,0 +1,10 @@
+			response = JsonResponse({
+		 		expectedAttr["ACTIVE"] : cms.active,
+		 		"success" : True,
+		 		})
```
Why's the name of the file `untitled`?

chinjieh commented on sendemail.py:
```diff
@@ -0,0 +1,16 @@
+
+@sched.scheduled_job('interval', minutes=1)
+def timed_job():
+    print('This job is run every one minutes.')
```
Does the `print` log to console?

chinjieh commented on sendemail.py:
```diff
@@ -0,0 +1,16 @@
+@sched.scheduled_job('interval', minutes=1)
+def timed_job():
+    print('This job is run every one minutes.')
+    global count, sched
```
Is `global` the only way? And why does `sched` need to be a global? It's not used in the code if I read correctly

chinjieh commented on sendemail.py:
```diff
@@ -0,0 +1,16 @@
+from apscheduler.schedulers.blocking import BlockingScheduler
```
Is `apscheduler` part of the default Python installation? Otherwise need to edit `requirements.txt`, remember!

chinjieh commented on emailApp/untitled:
```diff
@@ -0,0 +1,10 @@
+	 		 	time_difference = sgt_now() - cms.last_sent
+	 	time_difference_in_minutes = time_difference / timedelta(minutes=1)
+
+	 	if (cms.active ==True and time_difference_in_minutes>=30):
```
If I remember, Python is nice in that it can be written to make it very readable. So this can be rewritten as:

```
if cms.active and time_difference_in_minutes >= 30:
   # stuff
```

Why I like Python!

chinjieh reviewed (changes requested):
Good addition! But the `apscheduler` package thing should be checked before this can be approved.

Also, some Pythonic coding practice tips (if I remember them correctly)

jiaqi1993 committed:
Code Updated

Updated time_difference to time_exceeds_interval that returns a boolean
result if > timedelta(minutes =30)

Removed global variable in emailScheduler.py

Removed untitled txt file due to wrong commit.
